engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -563,6 +567,10 @@ protected void advanceExpunge(VMInstanceVO vm) throws ResourceUnavailableExcepti
         // Clean up volumes based on the vm's instance id
         volumeMgr.cleanupVolumes(vm.getId());
 
+        if (hostId != null && targets != null && targets.size() > 0) {
```

Comment:
You can use `hostId != null && CollectionUtils.isNotEmpty(targets)`.

Reply:
Done